rest-client/src/main/java/io/registry/client/RegistryRestService.java

```diff
@@ -0,0 +1,487 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.registry.client;
+
+import io.apicurio.registry.rest.beans.*;
+import io.apicurio.registry.service.RegistryService;
+import io.apicurio.registry.types.ArtifactType;
+import io.apicurio.registry.types.RuleType;
+import io.apicurio.registry.utils.IoUtil;
+import io.registry.client.callback.ResultCallback;
+import io.registry.client.service.ArtifactsService;
+import io.registry.client.service.IdsService;
+import io.registry.client.service.RulesService;
+import io.registry.client.service.SearchService;
+import okhttp3.OkHttpClient;
+import okhttp3.RequestBody;
+import okhttp3.ResponseBody;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+import javax.enterprise.inject.Vetoed;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * @author Carles Arnal <carnalca@redhat.com>
+ */
+@Vetoed // not a bean
+@Path("__dummy_hack")
+public class RegistryRestService implements RegistryService {
+
+    private final Retrofit retrofit;
+    private ArtifactsService artifactsService;
+    private RulesService rulesService;
+    private SearchService searchService;
+    private IdsService idsService;
+
+    public RegistryRestService() {
+        // hack + client side only
+        retrofit = null;
+    }
+
+    protected RegistryRestService(String baseUrl) {
+
+        retrofit = new Retrofit.Builder()
+                .baseUrl(baseUrl)
+                .addConverterFactory(JacksonConverterFactory.create())
+                .build();
+
+        initServices(retrofit);
+    }
+
+    protected RegistryRestService(String baseUrl, OkHttpClient okHttpClient) {
+
+        retrofit = new Retrofit.Builder()
+                .client(okHttpClient)
+                .addConverterFactory(JacksonConverterFactory.create())
+                .baseUrl(baseUrl)
+                .build();
+
+        initServices(retrofit);
+    }
+
+    public static RegistryService create(String baseUrl) {
+        return new RegistryRestService(baseUrl);
+    }
+
+    public static RegistryService create(String baseUrl, OkHttpClient okHttpClient) {
+        return new RegistryRestService(baseUrl, okHttpClient);
+    }
+
+    private void initServices(Retrofit retrofit) {
+        artifactsService = retrofit.create(ArtifactsService.class);
```

Comment:
So (latest) Retrofit "understands" jaxrs annotations?
(or it could be always, I just know that I used Retrofit's in my time -- ~2.5y ago)